crates/holochain_integrity_types/src/op.rs

```diff
@@ -81,77 +81,133 @@ pub enum Op {
-        /// The original [`Entry`] that is being deleted.
-        original_entry: Entry,
-    },
+    RegisterDelete(RegisterDelete),
     /// Registers a new [`Action`] on an agent source chain.
     /// This is the act of creating any [`Action`] and
     /// publishing it to the DHT.
-    RegisterAgentActivity {
-        /// The signed and hashed [`Action`] that is being registered.
-        action: SignedActionHashed,
-    },
+    RegisterAgentActivity(RegisterAgentActivity),
     /// Registers a link between two [`Entry`]s.
     /// This is the act of creating a [`Action::CreateLink`] and
     /// publishing it to the DHT.
     /// The authority is the entry authority for the base [`Entry`].
-    RegisterCreateLink {
-        /// The signed and hashed [`Action::CreateLink`] that registers the link.
-        create_link: SignedHashed<CreateLink>,
-    },
+    RegisterCreateLink(RegisterCreateLink),
     /// Deletes a link between two [`Entry`]s.
     /// This is the act of creating a [`Action::DeleteLink`] and
     /// publishing it to the DHT.
     /// The delete always references a specific [`Action::CreateLink`].
-    RegisterDeleteLink {
-        /// The signed and hashed [`Action::DeleteLink`] that registers the deletion.
-        delete_link: SignedHashed<DeleteLink>,
-        /// The link that is being deleted.
-        create_link: CreateLink,
-    },
+    RegisterDeleteLink(RegisterDeleteLink),
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, SerializedBytes)]
+#[cfg_attr(feature = "test_utils", derive(arbitrary::Arbitrary))]
+/// Stores a new [`Record`] in the DHT.
+/// This is the act of creating a new [`Action`]
+/// and publishing it to the DHT.
+/// Note that not all [`Action`]s contain an [`Entry`].
+pub struct StoreRecord {
+    /// The [`Record`] to store.
+    pub record: Record,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, SerializedBytes)]
+#[cfg_attr(feature = "test_utils", derive(arbitrary::Arbitrary))]
+/// Stores a new [`Entry`] in the DHT.
+/// This is the act of creating a either a [`Action::Create`] or
+/// a [`Action::Update`] and publishing it to the DHT.
+/// These actions create a new instance of an [`Entry`].
+pub struct StoreEntry {
+    /// The signed and hashed [`EntryCreationAction`] that creates
+    /// a new instance of the [`Entry`].
+    pub action: SignedHashed<EntryCreationAction>,
+    /// The new [`Entry`] to store.
+    pub entry: Entry,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, SerializedBytes)]
+#[cfg_attr(feature = "test_utils", derive(arbitrary::Arbitrary))]
+/// Registers an update from an instance of an [`Entry`] in the DHT.
+/// This is the act of creating a [`Action::Update`] and
+/// publishing it to the DHT.
+/// Note that the [`Action::Update`] stores an new instance
+/// of an [`Entry`] and registers it as an update to the original [`Entry`].
+/// This operation is only concerned with registering the update.
```

Comment:
I'm not sure what you mean?

Reply:
the last sentence alludes to there being more concerns around updates which are explicitly excluded from this function. as a relatively uninformed reader, i'm wondering what these other concerns around updates are and where they are handled. it would be nice to have a link or a short info about those.

Reply:
Oh it just means the update also creates a new entry but that's not what this is about

Reply:
i see. why not just mention that explicitly instead of hinting at an undefined something else?